test/Microsoft.AspNet.Mvc.Core.Test/ModelBinding/FormValueProviderFactoryTest.cs

```diff
@@ -49,6 +50,11 @@ private static ValueProviderFactoryContext CreateContext(string contentType)
             var context = new DefaultHttpContext();
             context.Request.ContentType = contentType;
 
+            if (context.Request.HasFormContentType)
+            {
+                context.Request.Form = new FormCollection(new Dictionary<string, string[]>());
+            }
```

Comment:
The Form will be read when the VP is created now instead of when the VP is used.